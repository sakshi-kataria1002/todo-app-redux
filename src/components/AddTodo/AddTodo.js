import {Grid, TextField, Button} from "@mui/material";
import { useState } from "react"
import axios from 'axios'

const AddTodo = () => {
    const [todo, setTodo] = useState({})

    const handleChange = (event) =>{
        const {name, value} = event.target
        setTodo({...todo, id:Math.random(), [name]:value})
        console.log(todo)
    }

    const handleSubmit = () => {
        axios.post("http://localhost:8080/add-Todo", todo)
        .then(response => console.log(response.data))
        //.catch(error => console.log(error))
    }

    return(
        <>
            <Grid container spacing={2}>
                <Grid item lg={12} mx={24}>
                    <TextField 
                        name="todoName" 
                        label="Todo Name" 
                        variant="filled"
                        onChange={handleChange} />
                </Grid>
                <Grid item lg={12} mx={24}>
                    <TextField 
                        name="todoDescription" 
                        label="Todo Description" 
                        variant="filled" 
                        onChange={handleChange} />
                </Grid>
                <Grid item lg={12} mx={29}>
                    <Button 
                        onClick={handleSubmit}
                        variant="contained"
                        >Add Todo
                    </Button>
                </Grid>
            </Grid>
        </>
    )
}

export default AddTodo