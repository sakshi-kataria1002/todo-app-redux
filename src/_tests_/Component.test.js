import React from 'react'
import {shallow} from 'enzyme'
import AddTodo from '../components/AddTodo/AddTodo'

describe('Testing Component Rendering', () => {
    it('should render correctly in debug mode', () => {
        const component = shallow(<AddTodo debug/>)
        expect(component).toMatchSnapshot()
    })
})