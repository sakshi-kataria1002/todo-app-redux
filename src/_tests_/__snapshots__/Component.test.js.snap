// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Component Rendering should render correctly in debug mode 1`] = `ShallowWrapper {}`;
